### PR TITLE
[Doc Canary] Updated EnumerableUtils documentation to more closely match Enumerable.

### PR DESCRIPTION
**Pull request by stavarotti**
Initially found an error with [indexOf](https://github.com/emberjs/ember.js/blob/v1.11.1/packages/ember-metal/lib/enumerable_utils.js#L64) docs but then proceeded to update the remaining methods to more closely match [Enumerable](https://github.com/emberjs/ember.js/blob/v1.11.1/packages/ember-runtime/lib/mixins/enumerable.js#L57).  I did not copy the docs verbatim but instead chose to improve a few areas.  Are these changes, if okay, worth while porting back to Enumerable?

### CODE REVIEW

File: packages/ember-metal/lib/enumerable_utils.js [hunk 1]
```diff
@@ -8,125 +8,183 @@ import {
...
- *
- * @method map
- * @param {Object} obj The object that should be mapped
- * @param {Function} callback The callback to execute
- * @param {Object} thisArg Value to use as this when executing *callback*
- *
- * @return {Array} An array of mapped values.
+  Creates and returns a new array with the results of calling the specified
+  `callback` on each element in the enumerable. This method corresponds to
+  `map()` defined in JavaScript 1.6.
```
Comment (mixonic) [inline packages/ember-metal/lib/enumerable_utils.js]: By 1.6 here, you mean EcmaScript 2015? I'd rather see "ES6" or ES2015" referenced if anything.

**Comment by mixonic** [comment]
@stavarotti I love it. Can you confirm that you've already applied the changes you mention in your overview to the enumerable utils?

File: packages/ember-metal/lib/enumerable_utils.js (hunk 1 above)
Comment (stavarotti) [inline packages/ember-metal/lib/enumerable_utils.js]: Updated.

**Comment by stavarotti** [comment]
@mixonic Confirmed.  I updated all methods in enumerable utils.  Are these changes worth porting back to Enumerable (I also noticed JavaScript 1.6 references in there)?  If so, I can open a separate PR to update the docs.

**Comment by stavarotti** [comment]
@mixonic Not sure why travis is failing (I rebased recently).  Nonetheless, can we get this one merged if all looks good?  I would like to update Enumerable's docs provided all changes here are green-lighted.

Thanks.

**Pull request closed without merging**